Battery_Checker.c

```diff
@@ -0,0 +1,91 @@
+#include <stdio.h>
+#include "Battery_Checker.h"
+
+BatteryStatus_tst BatteryStatus_st;
+
+float TempratureUnitConversion(float Temperature,TempratureUnit_ten TempUnit)
+{
+   float TempinCelcius = Temperature;
+  if(TempUnit == FAHRENHEIT)
+  {
+    TempinCelcius = ((Temperature - 32)*5/9);
+  }
+  return TempinCelcius;
+}
+int IsbatteryTempCheck(float Temprature)
+{
+   int ParamStatus = 0;
+  if(Temprature >= Min_Temp && Temprature <= Max_Temp) 
+  {
+      printf("Temperature out of range!\n");
+      if(Temprature < Min_Temp)
+      CheckTempforLowBreach();
+      else
+      CheckTempforHighBreach();
+        
+      ParamStatus = Not_Ok;
+  }
+  else
+  {
+    // CheckforWarning(Temprature);
+     ParamStatus = Ok;
+  }
+}
+
+int IsbatterySocCheck(float Soc)
+{
+   int ParamStatus = 0;
+  if(Soc > Min_Soc && Soc < Max_Soc) 
```

Comment:
this condition is for valid in range values... there should be test case to identify this bug...